enterprise/server/githubapp/githubapp.go

```diff
@@ -586,6 +592,98 @@ func (a *GitHubApp) GetAccessibleGitHubRepos(ctx context.Context, req *ghpb.GetA
 	return &ghpb.GetAccessibleReposResponse{RepoUrls: urls}, nil
 }
 
+func (a *GitHubApp) CreateRepo(ctx context.Context, req *repb.CreateRepoRequest) (*repb.CreateRepoResponse, error) {
+	tu, err := a.env.GetUserDB().GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if tu.GithubToken == "" {
+		return nil, status.UnauthenticatedError("github account link is required")
+	}
+
+	// Pick the right client based on the request (organization or user).
+	githubClient, token, err := a.newAuthenticatedClient(ctx, tu.GithubToken)
+	if req.Organization != "" {
+		githubClient, token, err = a.newInstallationClient(ctx, req.InstallationId)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new repository on github
+	repo, _, err := githubClient.Repositories.Create(ctx, req.Organization, &github.Repository{
+		Name:        github.String(req.Name),
+		Description: github.String(req.Description),
+		Private:     github.Bool(req.Private),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If we have a template, copy the template contents to the new repo
+	if req.Template != "" {
+		tmpDir := fmt.Sprintf("%s/template-repo/%d/%s/", os.TempDir(), req.InstallationId, req.Name)
+		cloneTemplate(tmpDir, token, req.Template, *repo.CloneURL)
+	}
```

Comment:
Outside the host ID file, we don't let the apps write files outside of the cache and (for remote_asset_api) a staging area. Could you use  "github.com/buildbuddy-io/buildbuddy/server/util/scratchspace" for this (just make the func tempDir() func public and replace os.TempDir() with it)?

Should be a drop in, and will guarantee your temp dir is on an ssd.

Reply:
done - i was going to ask you where i should put these temp files haha